### PR TITLE
[READY] Allow users to provide evidence for requirements

### PR DESCRIPTION
**Pull request by dkhgh**

### CODE REVIEW

File: db/migrate/20151027130308_create_evidences.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateEvidences < ActiveRecord::Migration
+  def change
+    create_table :evidences do |t|
+      t.belongs_to :user, index: true
+      t.belongs_to :requirement, index: true
+      t.datetime :valid_until, null: false
```
Comment (tarzan) [inline db/migrate/20151027130308_create_evidences.rb:6]: shouldn't the requirement take care of validity? i.e. a requirement with `within_months` just selects all evidence that is created during that timeframe in order to check whether it is met...

Comment (dkhgh) [inline db/migrate/20151027130308_create_evidences.rb:6]: no, I wanted to inherit it from the requirement so when a requirement changes the evidence doesn't :)

comparable to how when old drivers licenses are valid for 5 years, and from now on, newly issued ones are valid for 10 years, your old expiration date still holds :cop:

File: app/controllers/evidence_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+class EvidenceController < ApplicationController
```
Comment (dkhgh) [inline app/controllers/evidence_controller.rb]: this should be plural, right?
evidences?

File: config/routes.rb [hunk 3]
```diff
@@ -13,5 +13,7 @@
 
   root to: "pages#index"
   resources :users, only: [:index, :show, :edit, :update]
-  resources :requirements, only: [:index, :show, :create, :edit, :update, :destroy]
+  resources :requirements, only: [:index, :show, :create, :edit, :update, :destroy] do
+    resources :evidence, only: [:new, :create, :edit, :update]
```
Comment (dkhgh) [inline config/routes.rb]: should be plural

File: spec/controllers/evidence_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe EvidenceController, type: :controller do
```
Comment (dkhgh) [inline spec/controllers/evidence_controller_spec.rb]: should be plural

**Comment by GerwinV** [comment]
#6

**Pull request merged**